compiler/optimizer/OMRCFGSimplifier.cpp

```diff
@@ -333,6 +333,300 @@ bool OMR::CFGSimplifier::simplifyInstanceOfTestToCheckcast(bool needToDuplicateT
    return true;
    }
 
+// Looks for one of the patterns of the form:
+//   Pattern-1:
+//    ificmplt (i, 0) --------
+//        | (FallThrough)     |
+//    ificmplt (i, length) ======    
+//        |                   |  |
+//      throw  <--------------   |
+//        |                      |
+//      return <=================
+//
+//   Pattern-2:
+//    ificmplt (i, 0) -----------
+//        | (FallThrough)        |
+//    ificmpge (i, length) ------|   
+//        |                      |
+//      return                   |
+//        |                      |
+//       throw <-----------------
+//
+//   Pattern-3:
+//    ificmpge (i, 0) -----------
+//        | (FallThrough)        |
+//      throw                    |
+//        |                      |
+//    ificmpge (i, length) <-----
+//        |                      
+//      return
+//
+// That can be generated from the typical bound check code,
+//    if (i < 0 || i >= length)
+//       throw Exception();
+//    return i;
+//
+//  Or,
+//     if (i>=0 && i < length)
+//        return i;
+//     throw Exception();
+//
+// And tries to simplify them by replacing the ifcmp[lt/ge] blocks with a BNDCHK such that it becomes
+//
+//
+//  Simplification:
+//    BNDCHK (i, length)  ----(exp edge) --> goto
+//      |                                      |
+//    return                                   |
+//      |                                      |
```

Comment:
Again, let's only show CFG edges and remove the ones that are not needed.

Reply:
Removed the unnecessary edge, Reordered the goto block according to tree